### PR TITLE
updated docs relating to disk provisioning in vsphere environment

### PR DESCRIPTION
**Pull request by aleks-mariusz**
the documentation online relating to automatic disk provisioning
in a vmware environment (on the kubernetes orchestration platform)
did not include some critical parts.  i worked with _fm_ on slack
to determine the missing bits, and tightened security so documented
the exact privs that are needed.

### CODE REVIEW

**Comment by harsh-px** [comment]
Thanks @aleks-mariusz. We will review this soon.

**Comment by harsh-px** [comment]
@andreipope please review

File: content/cloud-references/auto-disk-provisioning/vsphere/_index.md [hunk 1]
```diff
@@ -52,10 +53,34 @@ Continue to [Install on-premise](/portworx-install-with-kubernetes/on-premise/)
 
 ### Step 3: Add env variables in the DaemonSet spec
 
-You will need to change the following sections in the downloaded spec to match your environment:
+You will need to change (or add) the following sections in the portworx DaemonSet section of the downloaded spec to match your environment:
```
Comment (andreipope) [inline content/cloud-references/auto-disk-provisioning/vsphere/_index.md]:
```suggestion
You will need to change (or add) the following sections in the Portworx DaemonSet section of the downloaded spec to match your environment:
```

File: content/cloud-references/auto-disk-provisioning/vsphere/_index.md [hunk 2]
```diff
@@ -52,10 +53,34 @@ Continue to [Install on-premise](/portworx-install-with-kubernetes/on-premise/)
...
+3. **VSPHERE_INSTALL_MODE**: "shared"
+4. **VSPHERE_USER**: should be a `valueFrom` with a secretKeyRef of `VSPHERE_USER` (referencing to the px-vsphere-secret defined above)
+5. **VSPHERE_PASSWORD**: should be a `valueFrom` with a secretKeyRef of `VSPHERE_PASSWORD` (referencing to the px-vsphere-secret defined above)
+
+Optionally, you may also need to specify these:
+
+6. **VSPHERE_VCENTER_PORT**: with the port number, if your vsphere services are not on the default port 443
+7. **VSPHERE_INSECURE**: if you are using self-signed certificates
+
+### Step 4: Permission your _vcenter-server-user_ appropriately
```
Comment (andreipope) [inline content/cloud-references/auto-disk-provisioning/vsphere/_index.md]: Looks better without italics:

```suggestion
### Step 4: Permission your vcenter-server-user appropriately
```

File: content/cloud-references/auto-disk-provisioning/vsphere/_index.md [hunk 3]
```diff
@@ -52,10 +53,34 @@ Continue to [Install on-premise](/portworx-install-with-kubernetes/on-premise/)
...
+
+Your _vcenter-server-user_ will need to either have the full Admin role, or for increased security, a custom-created role with the following minimum [vsphere privileges](https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.security.doc/GUID-FEAB5DF5-F7A2-412D-BF3D-7420A355AE8F.html):
```
Comment (andreipope) [inline content/cloud-references/auto-disk-provisioning/vsphere/_index.md]: I'd place the commas like this:

```suggestion
Your _vcenter-server-user_ will need to either have the full Admin role or, for increased security, a custom-created role with the following minimum [vsphere privileges](https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.security.doc/GUID-FEAB5DF5-F7A2-412D-BF3D-7420A355AE8F.html):
```

File: content/cloud-references/auto-disk-provisioning/vsphere/_index.md [hunk 4]
```diff
@@ -52,10 +53,34 @@ Continue to [Install on-premise](/portworx-install-with-kubernetes/on-premise/)
...
+  - Datastore.FileManagement
+  - Host.Local.ReconfigVM
+  - VirtualMachine.Config.AddExistingDisk
+  - VirtualMachine.Config.AddNewDisk
+  - VirtualMachine.Config.AddRemoveDevice
+  - VirtualMachine.Config.EditDevice
+  - VirtualMachine.Config.RemoveDisk
+  - VirtualMachine.Config.Settings
+
+The above permissions are in the format returned from the [govc utility](https://github.com/collabnix/govc) (which is in itself, useful for troubleshooting your vcenter-server-user access as well).
```
Comment (andreipope) [inline content/cloud-references/auto-disk-provisioning/vsphere/_index.md]: Don't think we need comma here:

```suggestion
The above permissions are in the format returned from the [govc utility](https://github.com/collabnix/govc) (which is in itself useful for troubleshooting your vcenter-server-user access as well).
```

**Comment by aleks-mariusz** [comment]
suggestions reviewed, added commit that takes onboard andreipope's recommendations

**Pull request merged**